tools/flecstan/CMakeLists.txt

```diff
@@ -3,16 +3,52 @@
 project(flecstan LANGUAGES CXX)
 cmake_minimum_required(VERSION 3.8)
 
+# Version
+# fixme Reformulate for newer cmake
+set(flecstan_VERSION_MAJOR 0)
+set(flecstan_VERSION_MINOR 0)
+set(flecstan_VERSION_PATCH 0)
+configure_file (
+  "src/flecstan-config.h.in"
+  "src/flecstan-config.h"
+)
+
 # Use cmake modules from cinch
 set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/../../cinch/cmake)
 
 # C++
+# fixme Replace hardcoded flags with something better
 set(CMAKE_CXX_STANDARD 17)
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pedantic -Wall")
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-rtti -O2 -s")
+if (NOT CMAKE_CXX_FLAGS)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pedantic -Wall")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-rtti -O2 -s")
+endif()
 
```

Comment:
You need to test to see if these compile, other compilers specify these differently.  https://cmake.org/cmake/help/v3.0/command/try_compile.html

Reply:
Right away I could just get rid of some; at least the -pedantic, -Wall, and -O2 are nice but not necessary. There appeared (last time I checked) to be some problems the Clang library had with RTTI. -s brought the executable down from around 500M to 50M (Clang is big).